src/java/org/apache/cassandra/audit/AuditLogEntry.java

```diff
@@ -204,9 +209,15 @@ public Builder(QueryState queryState)
                     source = InetAddressAndPort.getByAddressOverrideDefaults(addr.getAddress(), addr.getPort());
                 }
 
-                if (clientState.getUser() != null)
+                AuthenticatedUser authenticatedUser = clientState.getUser();
+                if (authenticatedUser != null)
                 {
-                    user = clientState.getUser().getName();
+                    user = authenticatedUser.getName();
+
+                    if (authenticatedUser.getMetadata() != null)
+                    {
+                        metadata = Map.copyOf(authenticatedUser.getMetadata());
```

Comment:
It's a protective copy since the `getMetadata` cannot be guaranteed to be non-mutable. I think we can move the creation of the `AuthenticatedUser` to use the `Map.of` method (`new AuthenticatedUser(role, MTLS, Map.of(METADATA_IDENTITY_KEY, identity))`. This in turn would give us a map that extends from type `AbstractImmutableMap`. `Map.copyOf` will avoid creating a new object and copying the map entries if the type is already an `AbstractImmutableMap`. So this should address the issue of increasing the memory usage.